src/samplers/powerspherical.jl

```diff
@@ -0,0 +1,27 @@
+# Sampler for Power Spherical
+
+struct PowerSphericalSampler{T <: Real} <: Sampleable{Multivariate,Continuous}
+    μ::Vector{T}
+    κ::T
+    dist_b::Beta
+    dist_u::HyperSphericalUniform
+end
+
+function _rand!(rng::AbstractRNG, spl::PowerSphericalSampler, x::AbstractVector)
+    z = rand(rng, spl.dist_b)
+    v = rand(rng, spl.dist_u)
+
+    t = 2 * z - 1
+    m = sqrt(1 - t ^ 2) * v'
+    y = [t; m]
+
+    x[1] = 1.
+    x[2:end] .= 0.
+    x .= x - spl.μ
+
+    normalize!(x)
+
+    x .= (-1) * (I(length(spl)) .- 2*x*x') * y
```

Comment:
Here's a completely non-allocating implementation:

```suggestion
    v = @views x[begin+1:end]
    uhat1 = 1 - spl.μ[begin]
    μv = @views spl.μ[begin+1:end]

    z = rand(rng, spl.dist_b)
    rand!(rng, spl.dist_u, v)

    t = 2 * z - 1

    v .*= sqrt(1 - t ^ 2)

    twouuᵀy1 = t * uhat1 - dot(v, μv)
    x[begin] = t - twouuᵀy1
    v .+= μv .* (twouuᵀy1 / uhat1)
    return x
```